src/coprocessor/dag/expr/mod.rs

```diff
@@ -457,6 +489,7 @@ fn filter<T: Into<Datum>>(x: Result<T>) -> Option<Result<Datum>> {
 impl Expression {
     pub fn eval(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Datum> {
         match *self {
+            Expression::ValueList(ref column) => unreachable!(),
```

Comment:
Why unreachable? When will it be used?

Reply:
Only for `IN` signature. We will write code like this:
```rust
match self.args[1] {  // args[1] is a list of datums.
    Expression::ValueList(ref datums) => {
        for d in datums.iter() {
            if self.args[0] == d {
                return true;
            }
        }
        return false;
    }
    _ => unreachable!(),
}
```
we can't write this logic in `eval` function, because what we want is `Vec<Datum>`, but `eval` returns `Datum`.

Reply:
Why not return `Err` instead?